frontend/src/app/patients/UploadPatients.tsx

```diff
@@ -0,0 +1,319 @@
+  const [buttonIsDisabled, setButtonIsDisabled] = useState(true);
+  const [selectedFacility, setSelectedFacility] = useState<Facility>();
+  const [file, setFile] = useState<File>();
+  const [errors, setErrors] = useState<
+    Array<FeedbackMessage | undefined | null>
+  >([]);
+  const [errorMessageText, setErrorMessageText] = useState("");
+  const [status, setStatus] = useState<
+    "submitting" | "complete" | "success" | "fail" | ""
+  >("");
+
+  const facilities = useSelector(
+    (state: any) => (state?.facilities as Facility[]) || []
+  );
+  const facility = selectedFacility || facilities[0] || { id: "", name: "" };
+
+  const onFacilitySelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selected = facilities.find((f) => f.id === e.target.value);
+    if (selected) {
+      setSelectedFacility(selected);
+    }
+  };
+
+  const handleFileChange = async (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    try {
+      if (!event?.currentTarget?.files?.length) {
+        return; //no files
+      }
+      const currentFile = event.currentTarget.files.item(0);
+      if (!currentFile) {
+        return;
+      }
+      setFile(currentFile);
+      setButtonIsDisabled(false);
+      setStatus("");
+    } catch (err: any) {
+      showError(`An unexpected error happened: '${err.toString()}'`);
+    }
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    setStatus("submitting");
+    setButtonIsDisabled(true);
+    setErrors([]);
+    setErrorMessageText("");
+
+    if (!file || file.size === 0) {
+      setStatus("fail");
+      setButtonIsDisabled(false);
+      setErrorMessageText("Invalid file");
+      return;
+    }
+    const facilityId = facilityAmount === "oneFacility" ? facility.id : "";
+    FileUploadService.uploadPatients(file, facilityId).then(async (res) => {
+      setStatus("complete");
+      setFile(undefined);
+      setButtonIsDisabled(true);
+
+      if (res.status !== 200) {
+        setStatus("fail");
+        setErrorMessageText(
+          "There was a server error. Your file has not been accepted."
+        );
+        if (res.body) {
+          const response = await res?.json();
+
+          if (response?.errors?.length) {
+            setErrorMessageText(
+              "Please resolve the errors below and upload your edited file."
+            );
+            setErrors(response.errors);
+          }
+        }
+      } else {
+        setStatus("success");
+      }
+    });
+  };
+
+  return (
+    <div className={"prime-edit-patient prime-home flex-1"}>
+      <div className={"grid-container margin-bottom-4 maxw-desktop-lg"}>
+        <div className="patient__header padding-bottom-0">
+          <AddPatientHeader />
+        </div>
+        <div className={"card padding-top-5 padding-bottom-7"}>
+          <span className={"font-sans-lg text-light line-height-225"}>
+            1. Setup your spreadsheet
```

Comment:
⚠️ re accessibility: These step elements need to be in an ordered list, not a span.